site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -56,4 +59,19 @@ public String save(Site site) {
         }
         return siteStore.saveEntity(site);
     }
+
+    /**
+     * Get complete sites list
+     * @return list of sites, never empty
+     * @throws InvariantException if the list from the store is empty
+     */
+    @GetMapping
+    public List<Site> getSites() {
+        final List<Site> sites = this.siteStore.geAllEntities();
+        if (sites.isEmpty()) {
+            throw new InvariantException(Services.NO_ROOT_SITE.message());
```

Comment:
1. Should we really return an error here? Is it because it's not a normal situation not to have any site?
2. Why specifically this error?
3. I think it's worth to add comments to describe the "why" in the code since it's probably not regular rest behavior.

Reply:
I went back and forth on this. My thinking was this is the site service so it is supposed to understand how sites work; if there is no root node (i.e. sites tree is empty), then the trial has not been initialized properly, so we should fail fast. Why this error? Having a root site should be an invariant for trials (after initialization), hence the exception name. I'll add a comment.